src/core/service.c

```diff
@@ -2578,16 +2580,18 @@ static void service_notify_cgroup_empty_event(Unit *u) {
 
         case SERVICE_START:
         case SERVICE_START_POST:
-                /* If we were hoping for the daemon to write its PID file,
-                 * we can give up now. */
-                if (s->pid_file_pathspec) {
+                if (s->type == SERVICE_NOTIFY)
+                        /* No chance of getting a ready notification anymore */
+                        service_enter_signal(s, SERVICE_FINAL_SIGTERM, SERVICE_FAILURE_PROTOCOL);
```

Comment:
The problem is not where the notifications come from, but that when PAM is used a cgroup_empty event is triggered around the same time the main process is executed, causing it to be killed instantly with the destruction of the cgroup. I am not sure what causes this event.

The way I see it, this code is correct and the extra cgroup_empty event is undesirable.